### DISCUSSION TITLE
Autolayout with ElkJS

### DISCUSSION THREAD

**Discussion by BOJIT** [open-source-labs/Svelvet#394] [Ideas]
Hi,

I've been looking into adding some auto-layout functionality to Svelvet using a library called [ElkJS](https://github.com/kieler/elkjs)

A quick demo of the functionality (from an old project) is shown below:
https://github.com/open-source-labs/Svelvet/assets/35117353/bedb55b4-c37c-4dd8-8223-2d7e3036515b

Would you be interested in this feature? If so I'm happy to raise a PR.

Thinking that this feature might be best implemented after native 'save' functionality is added: https://github.com/open-source-labs/Svelvet/discussions/336

Cheers,
James

**Comment by sradigan** [comment]
@BOJIT We are also using ElkJS with Svelvet.

Our layout code is here:
https://gitlab.com/c2-games/evidencing/cyber-atlas/-/blob/main/src/lib/proxiplay/nodeLayout.ts

We call the layout function before drawing in the svelvet canvas to generate the proper node information, then use svelte to render all the nodes:
https://gitlab.com/c2-games/evidencing/cyber-atlas/-/blob/main/src/routes/(app)/proxi-play/%2Bpage.svelte#L103

Our team would be interested in this being integrated into svelvet more natively as an optional reactive update to node position, size, or layout algorithm would be very useful going forward. It is difficult for us to handle new and remove nodes and edges between display changes without redrawing the canvas, which doesn't preserve user interaction modifications.

**Comment by briangregoryholmes** [comment]
Very interested. I'm prioritizing other things at the moment, but I'm completely open to looking into this when I have the time or in reviewing PRs.

If helpful, the position prop now features two way data binding in v8.0.4 and up.
